src/main/java/group1/mips_simulator/components/memParts/Cache.java

```diff
@@ -1,5 +1,93 @@
 package group1.mips_simulator.components.memParts;
 
+import group1.mips_simulator.Utility;
+import  group1.mips_simulator.components.Config;
+
+import java.util.*;
+
 public class Cache extends Memory{
-    public Cache(int size){super(size);}
+    /*
+    16 line, fully associative cache
+    Write Through (writes to cache, and immediately writes to memory (shouldnt need to implement separately since
+    in Java, variables names should be references to the same objects))
+
+    Cache is accessed usually by seeing if a certain data location is in a cache
+    (i.e. if the Word of a memory location has a tag that we extract, which should a line tag in cache.)
+
+    Fully associative cache means we can write to any line in cache, and we do so by writing over the oldest line first
+    (i.e. FIFO)
+     */
+
+    //data
+    int cacheSize = Config.CACHE_LINES;
+    Memory memory; //reference to memory that cache is linked to
+    //data structures
+    Queue<Short> cacheQueue; //going to use list as a queue for now
+    Map<Short,CacheLine> cacheLines; //<tag:block>
+
+
+    public Cache(Memory _memory){
+        //cache has a memory it references
+        memory = _memory;
+    }
+
+    public void addLine(short tag){
+        //get block of memory where tag is, create a cache block to be added to the cache
+        cacheQueue.add(tag); //add the new tag to the cache
+        cacheLines.put(tag, new CacheLine(tag, memory));
+    }
+
+    public void removeLine(){
+        // remove the oldest line from cache
+        Short tagToRemove = cacheQueue.poll(); //remove top CacheLine from Cache
+        cacheLines.remove(tagToRemove); //remove that tag from the lines Map
+    }
+
+
+    public CacheLine getLine(short dataAddr){
+        // get the full line of data at that line number
+        CacheLine returnLine;
+        short tag = getTag(dataAddr);
+
+        if(cacheLines.containsKey(tag))
+            returnLine = cacheLines.get(tag);
+        else{
+            if(cacheQueue.size() < cacheSize) { addLine(tag); }
+            else{ removeLine(); }
+            returnLine = cacheLines.get(tag);
+        }
```

Comment:
Let me try to translate this part to plain english:

If the cache does not contain the tag (line 54),
then check if the cache is full. 
If it is not full, add the tag (line 55)
If it is full, remove the oldest line (line 56)
Then get the tag from the cache. (line 57)

I think the add and remove lines are backwards/ confused here. It should be something like "If the cache does not contain the requested tag, then remove the oldest, and add the new line. Return that new line."